src/LoopDriver.php

```diff
@@ -7,20 +7,29 @@
     /**
      * Start the event loop.
      *
+     * The loop MUST continue to run until it is either stopped explicitly, no referenced watchers exist anymore, or an
+     * exception is thrown that cannot be handled. Exceptions that cannot be handled are exceptions thrown from an
+     * error handler or exceptions that would be passed to an error handler but non exists to handle them.
+     *
      * @return void
      */
     public function run();
 
     /**
      * Stop the event loop.
      *
+     * When an event loop is stopped, it continues with its current tick and exits the loop afterwards. Multiple calls
+     * to stop MUST be ignored and MUST NOT raise an exception.
+     *
      * @return void
      */
     public function stop();
 
     /**
      * Defer the execution of a callback.
      *
+     * The deferred callable MUST be executed in the next tick of the event loop.
```

Comment:
Otherwise, if we don't have this behavior, is there any concrete point (inside defer callbacks) to call defer()? In this case it would be just equivalent to calling the new callback at the end of the defer callback.